indico/modules/rb_new/controllers/backend/blockings.py

```diff
@@ -73,9 +75,40 @@ def _process(self, timeframe, my_rooms, mine):
         return jsonify(blockings_schema.dump(blockings).data)
 
 
-class RHRoomBlocking(RHRoomBookingBase):
+class RHRoomBlockingBase(RHRoomBookingBase):
     def _process_args(self):
         self.blocking = Blocking.get_one(request.view_args['blocking_id'])
 
+
+class RHRoomBlocking(RHRoomBlockingBase):
     def _process(self):
         return jsonify(blockings_schema.dump(self.blocking, many=False).data)
+
+
+class RHBlockingAction(RHRoomBlockingBase):
+    def _process_args(self):
+        RHRoomBlockingBase._process_args(self)
+        self.action = request.view_args['action']
+        self.blocked_room = (BlockedRoom.query
+                             .with_parent(self.blocking)
+                             .filter_by(id=request.view_args['blocked_room_id'])
```

Comment:
I think it would actually be cleaner to use the `room_id` here (especially considering the RESTful URL for this)